Keda.Cosmosdb.Scaler/src/Services/CosmosDBExternalScaler.cs

```diff
@@ -0,0 +1,173 @@
+﻿using System;
+using System.Threading.Tasks;
+using Google.Protobuf.Collections;
+using Grpc.Core;
+using Keda.CosmosDB.Scaler.Protos;
+using Microsoft.Azure.Documents.ChangeFeedProcessor;
+using Microsoft.Azure.Documents.ChangeFeedProcessor.DataAccess;
+using Microsoft.Azure.Documents.Client;
+using Microsoft.Extensions.Logging;
+using static Keda.CosmosDB.Scaler.Protos.ExternalScaler;
+
+namespace Keda.CosmosDB.Scaler.Services
+{
+    public class CosmosDBExternalScaler : ExternalScalerBase
+    {
+        private readonly ILogger _logger;
+
+        public CosmosDBExternalScaler(ILoggerFactory loggerFactory)
+        {
+            _logger = loggerFactory.CreateLogger<CosmosDBExternalScaler>();
+        }
+
+        public override async Task<IsActiveResponse> IsActive(ScaledObjectRef request, ServerCallContext context)
+        {
+            try
+            {
+                var trigger = CreateTriggerFromMetadata(request.ScalerMetadata);
+                var workToBeDone = await GetEstimatedWork(trigger);
+
+                bool isActive = workToBeDone > 0;
+
+                if (isActive)
+                {
+                    _logger.LogDebug(string.Format("Activating to 1 for cosmosDB account {0} database {1} collection {2}", trigger.AccountName, trigger.DatabaseName, trigger.CollectionName));
+                }
+
+                return new IsActiveResponse
+                {
+                    Result = isActive
+                };
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, "Error getting estimated work.");

```

Comment:
If we throw it, should we log it here? Otherwise, we could log it twice which is not ideal

Reply:
These logs emitted via the ILogger will eventually be consumed by our logging service eventually in a predecided format. Throw will only dump the exception stack to stdout stream of the pod. This is also the reason why I was not throwing the exception intially.

That being said, if you want we can remove the logger messages.

Reply:
Are you referring to the Azure Function logging service or something? Not sure if I'm following and how that comes into play since it should be standalone.

Just outputting to stderr is more than enough imo